### PR TITLE
Feature : Colorize files according to their status

### PR DESCRIPTION
**Pull request by justinehell**
- Get files status
- Colorize files according to their status

### CODE REVIEW

File: src/components/ModelizerTextView.vue [hunk 1]
```diff
@@ -115,28 +121,48 @@ function deleteFileTab(fileId) {
...
+function updateProjectFiles() {
+  let files;
+  return getProjectFiles(props.projectName).then((fileInformations) => {
+    files = fileInformations;
+    updateFileExplorer(fileInformations);
+    return getStatus(props.projectName);
+  }).then((filesStatus) => {
+    const fileInformations = files.map((file) => {
+      const statusResult = filesStatus.find(({ path }) => file.path === path);
+      if (statusResult) {
```
Comment (k-y0u) [inline src/components/ModelizerTextView.vue:158]: jump

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/components/ModelizerTextView.vue (hunk 1 above)
Comment (justinehell) [reply src/components/ModelizerTextView.vue:158]: done

File: src/components/ModelizerTextView.vue [hunk 2]
```diff
@@ -115,28 +121,48 @@ function deleteFileTab(fileId) {
...
+    setLastFileActive();
+  }
+}
+
+/**
+ * Get projects files and their status, then call function to make the update.
+ */
+function updateProjectFiles() {
+  let files;
+  return getProjectFiles(props.projectName).then((fileInformations) => {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/ModelizerTextView.vue:151]: done

File: src/components/ModelizerTextView.vue [hunk 3]
```diff
@@ -115,28 +121,48 @@ function deleteFileTab(fileId) {
...
-      fileTabArray.value = fileTabArray.value.filter(({ id }) => projectFilesIds.includes(id));
-
-      fileTabArray.value.forEach((file) => {
-        readProjectFile(props.projectName, { path: file.id })
-          .then(({ content }) => {
-            file.content = content;
-          });
+function updateFileExplorer(fileInformations) {
+  nodes.value = getTree(props.projectName, fileInformations);
+  const projectFilesIds = fileInformations.map((file) => file.path);
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/ModelizerTextView.vue:128]: ok

File: tests/unit/components/FileName.spec.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      it('Should match "label"', () => {
+        expect(wrapper.vm.props.label).toEqual('label');
+      });
+    });
+  });
+
+  describe('Test functions', () => {
+    describe('Test function: onUpdateFile', () => {
+      it('Should set files status if value is equal to props.path', async () => {
+        await wrapper.setProps({
```
[the remark this answers is not in the document]
Comment (justinehell) [reply tests/unit/components/FileName.spec.js:84]: yep removed

File: tests/unit/components/FileName.spec.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        wrapper.vm.filesStatus = 'status';
+
+        await wrapper.vm.onUpdateFile('path');
+
+        expect(wrapper.vm.filesStatus).toEqual('status');
+      });
+
+      it('Should not set files status if value is not equal to props.path', async () => {
+        await wrapper.setProps({
```
[the remark this answers is not in the document]
Comment (justinehell) [reply tests/unit/components/FileName.spec.js:99]: removed

File: tests/unit/components/FileName.spec.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+        expect(wrapper.vm.filesStatus).toEqual('');
+      });
+    });
+  });
+
+  describe('Test watcher: props.status', () => {
+    it('Should be triggered when props.status is updated', async () => {
+      wrapper.vm.props.status = '';
+      await wrapper.setProps({
```
[the remark this answers is not in the document]
Comment (justinehell) [reply tests/unit/components/FileName.spec.js:118]: remove l.117 and changed props.status value

File: src/composables/Project.js [hunk 7]
```diff
@@ -511,22 +511,26 @@ export async function deleteProjectFile(projectId, file, isFolder) {
 }
 
 /**
- * Get status of all files, except the unmodified files.
+ * Get the status of all files. If filePaths is defined, get the status of the files
+ * that strictly or partially match the given filePaths.
  * @param {String} projectId - Id of project.
+ * @param {String[]} filePaths - Limit the query to the given files and directories.
+ * @param {Function} filter - Filter the results to only those whose filepath matches a function..
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/composables/Project.js:518]: changed

File: src/composables/Project.js [hunk 8]
```diff
@@ -511,22 +511,26 @@ export async function deleteProjectFile(projectId, file, isFolder) {
...
+ * @param {String[]} filePaths - Limit the query to the given files and directories.
+ * @param {Function} filter - Filter to only return results whose filepath matches a given function.
  * @return {Promise<FileStatus[]>} All files status.
  */
-export async function getStatus(projectId) {
+export async function getStatus(projectId, filePaths, filter) {
   return git.statusMatrix({
     fs,
     dir: `/${projectId}`,
+    filepaths: filePaths,
```
Comment (Zorin95670) [inline src/composables/Project.js:525]: please rename `filePaths` to `filepaths` and make `filepaths,`

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/composables/Project.js (hunk 8 above)
Comment (justinehell) [reply src/composables/Project.js]: done

**Review by Zorin95670 — APPROVED** [review]

File: src/composables/FileExplorer.js [hunk 9]
```diff
@@ -16,21 +16,22 @@ export function createFolder(id, folder, name) {
 
 /**
  * Create and add a new file.
- * @param {String} id - Absolute path of file.
+ * @param {FileStatus[]} fileInformation - Absolute path and status of a file.
```
Comment (k-y0u) [inline src/composables/FileExplorer.js:19]: receiving an array of FileStatus but is named fileInformation, and without an S. Check the description also.

Comment (k-y0u) [reply src/composables/FileExplorer.js:19]: or maybe it's just the wrong type...

**Review by k-y0u — CHANGES_REQUESTED** [review]

**Review by k-y0u — DISMISSED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**